Running `pnpm tsc` will show that the ws module type is missing.

Running `pnpm tsc` will show a few type missing issues related to the ws module. You can add `@types/ws` to `programs/develop`.

> not able to repro but please push a PR with the proper fix

Have you encountered this error message on your side?
```
➜  extension.js git:(main) pnpm types

> extension@ types /Users/xinzhang/Documents/GitHub/extension.js
> pnpm tsc

programs/develop/webpack/plugin-reload/steps/create-web-socket-server/web-socket-server/broadcast-message.ts:1:23 - error TS2307: Cannot find module 'ws' or its corresponding type declarations.

1 import WebSocket from 'ws'
                        ~~~~

programs/develop/webpack/plugin-reload/steps/create-web-socket-server/web-socket-server/broadcast-message.ts:7:27 - error TS7006: Parameter 'client' implicitly has an 'any' type.

7   server.clients.forEach((client) => {
                            ~~~~~~

programs/develop/webpack/plugin-reload/steps/create-web-socket-server/web-socket-server/message-dispatcher.ts:2:23 - error TS2307: Cannot find module 'ws' or its corresponding type declarations.

2 import WebSocket from 'ws'
                        ~~~~

programs/develop/webpack/plugin-reload/steps/create-web-socket-server/web-socket-server/message-dispatcher.ts:11:27 - error TS7006: Parameter 'client' implicitly has an 'any' type.

11   server.clients.forEach((client) => {
                             ~~~~~~

programs/develop/webpack/plugin-reload/steps/create-web-socket-server/web-socket-server/start-server.ts:3:23 - error TS2307: Cannot find module 'ws' or its corresponding type declarations.

3 import WebSocket from 'ws'
                        ~~~~

programs/develop/webpack/plugin-reload/steps/create-web-socket-server/web-socket-server/start-server.ts:61:37 - error TS7006: Parameter 'ws' implicitly has an 'any' type.

61   webSocketServer.on('connection', (ws) => {
                                       ~~

programs/develop/webpack/plugin-reload/steps/create-web-socket-server/web-socket-server/start-server.ts:64:21 - error TS7006: Parameter 'error' implicitly has an 'any' type.

64     ws.on('error', (error) => {
                       ~~~~~

programs/develop/webpack/plugin-reload/steps/create-web-socket-server/web-socket-server/start-server.ts:68:23 - error TS7006: Parameter 'msg' implicitly has an 'any' type.

68     ws.on('message', (msg) => {
                         ~~~


Found 8 errors in 3 files.

Errors  Files
     2  programs/develop/webpack/plugin-reload/steps/create-web-socket-server/web-socket-server/broadcast-message.ts:1
     2  programs/develop/webpack/plugin-reload/steps/create-web-socket-server/web-socket-server/message-dispatcher.ts:2
     4  programs/develop/webpack/plugin-reload/steps/create-web-socket-server/web-socket-server/start-server.ts:3
 ELIFECYCLE  Command failed with exit code 2.
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -22,6 +22,12 @@ jobs:
       run: pnpm install
     - name: Run compiler
       run: pnpm compile
+    - name: Run format
+      run: pnpm format
+    - name: Run types
+      run: pnpm types
+    - name: Run lint
+      run: pnpm lint
     - name: Run tests
       run: pnpm test:coverage
     - name: Upload results to Codecov

--- a/programs/develop/package.json
+++ b/programs/develop/package.json
@@ -81,6 +81,7 @@
     "@types/loader-utils": "^2.0.6",
     "@types/node": "^22.5.3",
     "@types/webextension-polyfill": "^0.10.7",
+    "@types/ws": "^8.5.12",
     "jest": "^29.7.0",
     "ts-jest": "^29.1.2",
     "tsup": "^8.0.1",

